pkg/cloudprovider/cloudprovider.go

```diff
@@ -100,6 +106,44 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
+func (c *CloudProvider) IsNodeDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, node *v1.Node) (bool, error) {
+	instanceTypes, err := c.GetInstanceTypes(ctx, provisioner)
+	if err != nil {
+		return false, fmt.Errorf("getting instanceTypes, %w", err)
+	}
+	nodeInstanceType := lo.Filter(instanceTypes, func(instType *cloudprovider.InstanceType, _ int) bool {
+		return instType.Name == node.Labels[v1.LabelInstanceTypeStable]
+	})
+	if len(nodeInstanceType) == 0 {
+		return false, fmt.Errorf("getting nodeInstanceType")
+	}
+	nodeTemplate, err := c.resolveNodeTemplate(ctx, provisioner.Spec.Provider, provisioner.Spec.ProviderRef)
+	if err != nil {
+		return false, fmt.Errorf("resolving node template, %w", err)
+	}
+	if nodeTemplate.Spec.LaunchTemplateName != nil {
+		return false, fmt.Errorf("using a custom Launch Template which is deprecated")
+	}
+	amis, err := c.amiProvider.Get(ctx, nodeTemplate, nodeInstanceType, amifamily.GetAMIFamily(nodeTemplate.Spec.AMIFamily, &amifamily.Options{}))
+	if err != nil {
+		return false, fmt.Errorf("getting amis, %w", err)
+	}
+	nodeAmi, ok := node.Labels[v1alpha1.LabelInstanceAMIID]
+	if !ok {
+		instanceID, err := utils.ParseInstanceID(node)
```

Comment:
Since we're doing the fallback, can you add a suite test for when the node does not have the AMI ID?

Reply:
Done, but not sure if my implementation is 100% right. You might have to look.